web/src/components/tasks/business-formation/contacts/Addresses.tsx

```diff
@@ -118,57 +120,72 @@ export const Addresses = <T extends FormationMember | FormationIncorporator>(
             return (
-              <tr className="margin-bottom-1" key={index} data-testid={`${props.fieldName}-${index}`}>
-                <td className="break-word">{it.name}</td>
-                <td className="break-word">{formatAddress(it)}</td>
-                {"signature" in it ? (
-                  <td className="padding-y-0">
-                    {" "}
-                    {renderSignatureColumn({
-                      onChange: (event) => {
-                        return handleSignerCheckbox(event, index);
-                      },
-                      checked: it.signature,
-                      fieldName: props.fieldName,
-                      index,
-                    })}
+              <Fragment key={index}>
+                <tr className="margin-bottom-1" key={index} data-testid={`${props.fieldName}-${index}`}>
+                  <td className="break-word">{it.name}</td>
+                  <td className="break-word">{formatAddress(it)}</td>
+                  {"signature" in it ? (
+                    <td className="padding-y-0">
+                      {" "}
+                      {renderSignatureColumn({
+                        onChange: (event) => {
+                          return handleSignerCheckbox(event, index);
+                        },
+                        checked: it.signature,
+                        fieldName: props.fieldName,
+                        index,
+                      })}
+                    </td>
+                  ) : (
+                    <></>
+                  )}
+                  <td className="display-inline-flex">
+                    <div>
+                      <IconButton
+                        aria-label="edit"
+                        onClick={(): void => {
+                          setEditIndex(index);
+                          setModalOpen(true);
+                        }}
+                        className="usa-button usa-button--unstyled"
+                      >
+                        <Icon className="usa-icon--size-3">edit</Icon>
+                      </IconButton>
+                    </div>
+                    <div className="margin-x-1 border-1px border-base-light" />
+                    <div>
+                      <IconButton
+                        aria-label="delete"
+                        onClick={(): void => deleteAddress(index)}
+                        className="usa-button usa-button--unstyled"
+                      >
+                        <Icon className="usa-icon--size-3">delete</Icon>
+                      </IconButton>
+                    </div>
                   </td>
+                </tr>
+                {doesFieldHaveError(props.fieldName) && "signature" in it && !it.signature ? (
+                  <tr key={`error-${index}`}>
+                    <td colSpan={4} className="text-error-dark text-bold">
+                      {Config.formation.fields.signers.errorBannerCheckbox}
```

Comment:
[sand/dust] 

Should this {Config.formation.fields.signers.errorBannerCheckbox} be inside of getErrorStateForFields()? 

Are we trying to consolidate all our error logic in there? Or is that unecessary in this case for some reason?

Reply:
Actually, its probably fine as is. We use both patterns

Reply:
Cool...leaving!